File: 백준/Bronze/25304. 영수증/영수증.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split(/\s/);

const price = +input[0];
const product = input[1];
const recipeArr = input.slice(2, price);
let result = 0;

for (let i = 0; i < recipeArr.length; i += 2) {
  result += recipeArr[i] * recipeArr[i + 1];
}

result === price ? console.log("Yes") : console.log("No");

File: 백준/Bronze/25304. 영수증/영수증.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int totalMoney = sc.nextInt();
        int calculateMoney = 0;
        int n = sc.nextInt();
        for(int i=0; i<n; i++) {
            int price = sc.nextInt();
            int cnt = sc.nextInt();
            calculateMoney += price * cnt;
        }
        String result = calculateMoney == totalMoney ? "Yes" : "No";
        System.out.println(result);
    }
}